tools/tiled_ensemble/ensemble_functions.py

```diff
@@ -0,0 +1,259 @@
+"""Functions used to train and use ensemble of models."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import copy
+import logging
+import os
+import warnings
+from typing import Any, Dict, List, Tuple
+
+from omegaconf import DictConfig, ListConfig, OmegaConf
+from pytorch_lightning import Callback
+from pytorch_lightning.callbacks import ModelCheckpoint
+from tools.tiled_ensemble.ensemble_tiler import EnsembleTiler
```

Comment:
I think that code should be inside anomalib, that would also fit better with running the script as right now root needs to be added to pytonpath to work (just the way python works if you have source in same root as script).

About naming, my idea was that the method is called tiled ensemble, but tiler class for this method is called ensemble tiler, but can be renamed to tiled ensemble tiller. I think it'd be good to hear @djdameln thoughts on this.

Reply:
@ashwinvaidya17 it would be great to also hear your opinion on this.

Reply:
I agree. These methods can go inside `anomalib`. We can consider having a folder called pipelines/orchestration in anomalib. Maybe later on we can move these into the trainer folder (anomalib 1.0). We can collect exerimentation utilities here as well (like HPO). This will also make all these utilities available with `pip install`. I didn't go through it in detail but from my cursory look, we can move most of the code within anomalib and keep only the entry point. In Anomalib 1.0 we can consider replacing the entry point with `anomalib ensemble fit/test`.
Other than that I have a few ideas for refactoring but that might not be part of the scope of this project and is aligned with the anomalib 1.0 design. The rest of the code looks very clean. Good job!